python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/vlan-interfaces.j2

```diff
@@ -315,6 +315,65 @@ interface {{ vlan_interface.name }}
 {%     if vlan_interface.isis_network_point_to_point is arista.avd.defined(true) %}
    isis network point-to-point
 {%     endif %}
+{%     if vlan_interface.isis_authentication_mode is arista.avd.defined %}
+{%         set auth_cli = "isis authentication mode" ~ " " ~ vlan_interface.isis_authentication_mode %}
+{%         if vlan_interface.isis_authentication_mode == "sha" and vlan_interface.isis_authentication_mode_sha is arista.avd.defined %}
+{%             if vlan_interface.isis_authentication_mode_sha.both.key_id is arista.avd.defined %}
+{%                 set auth_cli = auth_cli ~ " key-id " ~ vlan_interface.isis_authentication_mode_sha.both.key_id %}
+{%                 if vlan_interface.isis_authentication_mode_sha.both.rx_disabled is arista.avd.defined %}
+{%                     set auth_cli = auth_cli ~ " rx-disabled" %}
+{%                 endif %}
+   {{ auth_cli }}
+{%             else %}
+{%                 if vlan_interface.isis_authentication_mode_sha.level_1.key_id is arista.avd.defined %}
+{%                     if vlan_interface.isis_authentication_mode_sha.level_1.rx_disabled is arista.avd.defined %}
+{%                         set auth_cli = auth_cli ~ " rx-disabled" %}
+{%                     endif %}
+{%                     set auth_cli = auth_cli ~ " key-id " ~ vlan_interface.isis_authentication_mode_sha.level_1.key_id ~ " level-1" %}
+   {{ auth_cli }}
+{%                 endif %}
+{%                 if vlan_interface.isis_authentication_mode_sha.level_2.key_id is arista.avd.defined %}
```

Comment:
this should be in `elif`
if both level-1 and level2 are given in input, it generates invalid config.

```
- name: Vlan341
    description: isis authentication sha
    isis_authentication_mode: sha
    isis_authentication_mode_sha:
      level_1:
        key_id: 1010
      level_2:
        key_id: 12
```
```
isis authentication mode sha key-id 1010 level-1
isis authentication mode sha key-id 1010 level-1 key-id 12 level-2
```

The above config is invalid

Reply:
Fixed.